tests/publiccloud/create_aistack_env.pm

```diff
@@ -181,13 +226,21 @@ sub test_openwebui_service {
     set_var('OPENWEBUI_IP', "$ipaddr");
     record_info("Added $ipaddr to /etc/hosts with hostname $host_name");
 
-    # connect open-webui service
-    my $curl_cmd = "curl -v -k https://$host_name";
-    my $curl_result = script_run($curl_cmd);
-    if ($curl_result == 0) {
-        record_info("Successfully connected to the open-webui service at $curl_cmd \n");
+    # get endpoints
+    my $endpoint_cmd = "kubectl get endpoints $sr_name -n $namespace -o=jsonpath='{.subsets[*].addresses[*].ip}'";
+    my $endpoint_result = script_run($endpoint_cmd);
+    if ($endpoint_result eq "") {
```

Comment:
Sorry but `script_run()` returns bash-like exit codes: 0 when ok, 1 or else when error occurred.
Then `result eq ""` makes no sense as check.
To catch error here, try:
```suggestion
    if ($endpoint_result != 0) {
```
But I suggest inverted logic: first run then-block `... == 0`, otherwise `else die "No .."`

Reply:
Ok, but, other than above `script_output`, I suggest:
```suggestion
    unless ($endpoint_result) { ...
```
looking for empty output string

Reply:
modified to 
`   unless ($endpoint_result) {
        die "No healthy endpoints found for the open-webui service in $namespace\n";
    } else {
        # connect to open-webui service
        my $curl_cmd = "curl --output /dev/null --silent --head --write-out \"%{http_code}\n\" -k -L https://$host_name";
        my $curl_result = script_output($curl_cmd);
        record_info("HTTP code: $curl_result \n");

        if ($curl_result == 200) {
            record_info("Successfully connected to the open-webui service at $curl_cmd \n");
        } else {
            die "Received unexpected HTTP error code $curl_result for $curl_cmd\n";
        }
    }

`

Reply:
https://openqa.suse.de/tests/16209439#details Fixed. here is the VR
` if (!$endpoint_result) {
        die "No healthy endpoints found for the open-webui service in $namespace\n";
    } else {
        # connect open-webui service
        my $curl_cmd = "curl --output /dev/null --silent --head --write-out \"%{http_code}\n\" -k -L https://$host_name";
        my $curl_result = script_output($curl_cmd);
        record_info("http code: $curl_result \n");
        if ($curl_result == 200) {
            record_info("Successfully connected to the open-webui service at $curl_cmd \n");
        } else {
            die "Received unexpected HTTP error code $curl_result for $curl_cmd\n";
        }
    }
`